frontend/src/pages/NewRun.tsx

```diff
@@ -322,6 +331,34 @@ class NewRun extends Page<{}, NewRunState> {
         await this.showPageError(`Error: failed to retrieve original run: ${originalRunId}.`, err);
         logger.error(`Failed to retrieve original run: ${originalRunId}`, err);
       }
+    } else if(embeddedPipelineRunId) {
```

Comment:
nit: Does it make sense for this logic to move into the `_prepareFormFromClone`? (we can rename that method accordingly, maybe `_prepareFormFromRun`).
Alternatively, we can add a new method, to keep this part of the code about branching conditions rather than branch implementations.

Reply:
Yes, I think that makes a lot of sense. Created a new function: `_prepareFormFromEmbeddedPipeline`.

Very little logic is shared with `_prepareFormFromClone`, so it makes sense to keep them separate.